### PR TITLE
feat(api,kikan): host-header allowlist — DNS-rebinding defense (#518)

### LINKED ISSUE #518: Host-header allowlist tower layer (kikan-provided, DNS-rebinding defense)
## Summary

Implement a kikan-provided `tower::Layer` that rejects any HTTP request whose `Host` header is not in a compile-time allowlist (`127.0.0.1:*`, `localhost:*`). Applied at the top of both `mokumo_api::build_app()` (data plane) and `mokumo_api::build_unix_admin_app()` (control plane subset).

This is **commitment #1** from the 2026-04-13 HTTP-EVERYWHERE council decision. It is the load-bearing DNS-rebinding defense for HTTP-over-loopback. Without this layer, a malicious webpage can use DNS rebinding (per [CVE-2018-1000817](https://nvd.nist.gov/vuln/detail/CVE-2018-1000817), [CVE-2018-5702](https://nvd.nist.gov/vuln/detail/CVE-2018-5702), [CVE-2020-8562](https://nvd.nist.gov/vuln/detail/CVE-2020-8562)) to bypass the Same-Origin Policy and reach the loopback listener from the user's browser.

## Scope

- New module `crates/kikan/src/middleware/host_allowlist.rs`
- `HostHeaderAllowList` struct implementing `tower::Layer<S>` and `tower::Service<Request>`
- Compile-time allowlist: `127.0.0.1`, `localhost`, any IPv6 loopback equivalent
- In `DeploymentMode::ReverseProxy`, the allowlist can be extended at runtime by the operator (proxy hostname allowed)
- Rejected requests return `403 Forbidden` with a minimal body
- Applied by default in `mokumo_api::build_app()` and `build_unix_admin_app()` — **not opt-in** (a future steward removing the layer should be flagged by a CI check)

## Acceptance criteria

- [ ] `HostHeaderAllowList` tower layer exists in `crates/kikan/src/middleware/`
- [ ] Applied at the top of both `build_app()` and `build_unix_admin_app()` routers
- [ ] Integration test: request to `http://127.0.0.1:6565/api/healthz` succeeds (Host header `127.0.0.1`)
- [ ] Integration test: request with `Host: evil.com` receives 403
- [ ] Reviewer-visible: any PR that removes the layer fails CI (invariant check)
- [ ] `DeploymentMode::ReverseProxy` allows operator-configured proxy hostname

## Refs

- [[decisions/mokumo/adr-tauri-http-not-ipc]] 2026-04-13 amendment (commitment #1)
- [[decisions/mokumo/adr-kikan-deployment-modes]]
- [[education/kikan-foundations/src/03-architectural-decisions/05-tauri-ipc-pivot]] — worked example

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

- Adds `HostHeaderAllowList` tower layer in `crates/kikan` — rejects requests whose `Host` header is not `127.0.0.1`, `localhost`, or `[::1]` (any port) with 403 before any handler, auth, or tracing
- Fail closed: missing Host → 403, multiple Host headers → 403 (smuggling defense)
- Applied as outermost `.layer()` in `build_app_inner()` — kikan's first production deps (`tower 0.5` + `http 1`)
- 403 body matches project error contract: `{"code":"HOST_NOT_ALLOWED","message":"Host header not allowed","details":null}`

## Test plan

- [x] 19 unit tests in `crates/kikan` covering CQO edge case matrix: missing Host, multiple headers, case-insensitive, IPv6, port stripping, `127.0.0.1.evil.com`, `localhost.evil.com`, percent-encoded
- [x] 2 integration tests in `services/api`: `host_evil_rejected` (403 + body contract), `host_loopback_accepted` (200 on `/api/health`)
- [x] BDD feature file with 13 scenarios
- [x] Hurl smoke test (`tests/api/security/host-allowlist.hurl`)
- [x] Clippy clean on both crates
- [ ] CI pipeline passes

Closes #518

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/kikan/src/middleware/host_allowlist.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        let host_values: Vec<_> = req.headers().get_all(HOST).into_iter().collect();
+
+        if host_values.len() != 1 {
+            return Box::pin(std::future::ready(Ok(build_rejection())));
+        }
+
+        let header_val = match host_values[0].to_str() {
+            Ok(v) => v,
+            Err(_) => return Box::pin(std::future::ready(Ok(build_rejection()))),
+        };
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/kikan/src/middleware/host_allowlist.rs]: Fixed in db30e98 — switched to iterator pattern matching to avoid the Vec allocation on the hot path.
